### PR TITLE
Fix unused-var, shadowed variable warnings and integration test timeouts and descendent chains

### PR DESCRIPTION
**Pull request by dcousens**
Related to https://github.com/feross/standard/issues/636#issuecomment-270815470

### CODE REVIEW

File: src/templates/index.js [hunk 1]
```diff
@@ -53,7 +53,7 @@ function classifyWitness (script, allowIncomplete) {
   var chunks = decompile(script)
 
   if (pubKeyHash.input.check(chunks)) return types.P2WPKH
-  if (scriptHash.input.check(chunks)) return types.P2WSH
+  if (scriptHash.input.check(chunks, allowIncomplete)) return types.P2WSH
```
Comment (dcousens) [inline src/templates/index.js]: This... needs tests

**Comment by dcousens** [comment]
rebased - will merge if good (may have integration tests fail until https://github.com/bitcoinjs/bitcoinjs-lib/pull/786)

File: src/templates/index.js (hunk 1 above)
Comment (dcousens) [inline src/templates/index.js]: https://github.com/bitcoinjs/bitcoinjs-lib/issues/755

**Comment by dcousens** [comment]
Merged the test fixes into this as it was frustrating having fixes in both branches.

**Comment by fanatid** [comment]
ack

**Pull request merged**